From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 23 May 2018 21:14:00 +0200
Subject: [PATCH] Remove query operation from tunefs

tunefs.ocfs2/op_query.c uses glibc's printf extensions that are not
available in the musl libc. I didn't find any standalone implementation
and it's not trivial to implement them, so we must remove this feature
for now.

--- a/tunefs.ocfs2/Makefile
+++ b/tunefs.ocfs2/Makefile
@@ -42,7 +42,6 @@
 	op_cloned_volume		\
 	op_features			\
 	op_list_sparse_files		\
-	op_query			\
 	op_reset_uuid			\
 	op_resize_volume		\
 	op_set_label			\
--- a/tunefs.ocfs2/ocfs2ne.c
+++ b/tunefs.ocfs2/ocfs2ne.c
@@ -89,7 +89,6 @@
 
 
 extern struct tunefs_operation list_sparse_op;
-extern struct tunefs_operation query_op;
 extern struct tunefs_operation reset_uuid_op;
 extern struct tunefs_operation features_op;
 extern struct tunefs_operation resize_volume_op;
@@ -498,17 +497,6 @@
 	.opt_handle	= handle_answer,
 };
 
-static struct tunefs_option query_option = {
-	.opt_option	= {
-		.name		= "query",
-		.val		= 'Q',
-		.has_arg	= 1,
-	},
-	.opt_help	= "-Q|--query <query-format>",
-	.opt_handle	= &generic_handle_arg,
-	.opt_op		= &query_op,
-};
-
 static struct tunefs_option list_sparse_option = {
 	.opt_option	= {
 		.name	= "list-sparse",
@@ -652,7 +640,6 @@
 	&resize_volume_option,
 	&reset_uuid_option,
 	&journal_option,
-	&query_option,
 	&list_sparse_option,
 	&mount_type_option,
 	&backup_super_option,
